src/clr-angular/accordion/accordion-panel.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {
+  Component,
+  ChangeDetectionStrategy,
+  Input,
+  EventEmitter,
+  Output,
+  SimpleChanges,
+  Inject,
+  OnInit,
+  OnChanges,
+} from '@angular/core';
+import { Observable } from 'rxjs';
+import { tap } from 'rxjs/operators';
+
+import { UNIQUE_ID_PROVIDER, UNIQUE_ID } from '../utils/id-generator/id-generator.service';
+import { ClrCommonStrings } from '../utils/i18n/common-strings.interface';
+import { AccordionService } from './providers/accordion.service';
+import { AccordionStatus } from './enums/accordion-status.enum';
+import { panelAnimation } from './utils/animation';
+import { IfExpandService } from '../utils/conditional/if-expanded.service';
+import { AccordionPanelModel } from './models/accordion.model';
+
+@Component({
+  selector: 'clr-accordion-panel',
+  templateUrl: './accordion-panel.html',
+  host: { '[class.clr-accordion-panel]': 'true' },
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  animations: panelAnimation,
+  providers: [IfExpandService, UNIQUE_ID_PROVIDER],
+})
+export class ClrAccordionPanel implements OnInit, OnChanges {
+  @Input('clrAccordionPanelOpen') panelOpen = false;
+  @Output('clrAccordionPanelOpenChange') panelOpenChange = new EventEmitter<boolean>();
+
+  panel: Observable<AccordionPanelModel>;
+  focusHeader = false;
+  readonly AccordionStatus = AccordionStatus;
+
+  constructor(
+    public commonStrings: ClrCommonStrings,
+    private accordionService: AccordionService,
+    private ifExpandService: IfExpandService,
+    @Inject(UNIQUE_ID) public id: string
+  ) {}
+
+  ngOnInit() {
+    this.panel = this.accordionService.getPanelChanges(this.id).pipe(tap(panel => this.emitPanelChange(panel)));
```

Comment:
I am not very knowledgeable on this one. Does it do the unsubscriptions itself?

Reply:
It is managed by the `async` pipe in the template